src/transformers/models/llama/modeling_llama.py

```diff
@@ -1251,7 +1258,12 @@ def prepare_inputs_for_generation(
         if inputs_embeds is not None and past_key_values is None:
             model_inputs = {"inputs_embeds": inputs_embeds}
         else:
-            model_inputs = {"input_ids": input_ids}
+            # The `contiguous()` here is necessary to have a static stride during (non-speculative) decoding. torchdynamo otherwise
+            # recompiles graphs as the stride of the inputs is a guard.
+            # TODO: We don't really need to handle the input_ids here, and this contiguous() call could be removed if we were
+            # simply using GenerationMixin.greedy_search `next_tokens` variable directly (which is already contiguous), instead of
+            # doing a torch.cat + then slice.
+            model_inputs = {"input_ids": input_ids.contiguous()}
```

Comment:
As per the comment. If `input_ids` is not contiguous, its stride is different at each forward call in generate, which triggers a recompilation at every step in the loop. This is very slow.

Ideally we would want to just use `next_tokens` instead of having this `contiguous()` call, but let's do a proper refactor in an other PR

Reply:
Yep, next token would be less "costly" but I think we need all of them for speculative? Anyways noticed that as well when I was compiling, just let it be at the time!